OSX linking error when making demo: "Undefined symbol __SDL_main "

``` sh
$ make demos
rustc -L . demo/demo.rc -o demos
demo/video.rs:3:4: 3:11 warning: unused import [-W unused-imports (default)]
demo/video.rs:3 use std::rt;
                    ^~~~~~~
error: linking with `cc` failed with code 1
note: cc arguments: -L/usr/local/rust-master/lib/rustc/x86_64-apple-darwin/lib -m64 -o demos demos.o -L/usr/local/rust-master/lib/rustc/x86_64-apple-darwin/lib -lstd-6c65cf4b443341b1-0.8-pre -L. -lsdl-95460b1b072257-0.3.0 -lSDL_image -lSDL_mixer -lSDL -lSDLmain -framework Foundation -framework AppKit -L. -L/Users/james/Code/rust-sdl/.rust -L/Users/james/Code/rust-sdl -Wl,-no_compact_unwind -lmorestack -lrustrt -Wl,-rpath,@executable_path/../../../../usr/local/rust-master/lib/rustc/x86_64-apple-darwin/lib -Wl,-rpath,@executable_path/. -Wl,-rpath,/usr/local/rust-master/lib/rustc/x86_64-apple-darwin/lib -Wl,-rpath,/Users/james/Code/rust-sdl/.
note: ld: warning: directory not found for option '-L/Users/james/Code/rust-sdl/.rust'
Undefined symbols for architecture x86_64:
  "_SDL_main", referenced from:
      -[SDLMain applicationDidFinishLaunching:] in libSDLmain.a(SDLMain.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error: aborting due to previous error
make: *** [demos] Error 101
```

--- a/src/sdl-demo/sdl_main.rs
+++ b/src/sdl-demo/sdl_main.rs
@@ -1,0 +1,70 @@
+//! Specialized demo of SDL for use on platforms that really want to
+//! initialize and own the main routine and thread.  This is mean to be
+//! linked with object code compiled from one of the SDL_main variants
+//! that one can find beneath <SDL-distribution>/src/main/.
+//!
+//! For example on Mac OS X on can build an runnable program from this
+//! with something along the lines of:
+//!
+//!   rustc sdl_main.rs -L. -L<SDL-lib> -C link-args=" -I<SDL-include>/SDL   \
+//!      -framework CoreFoundation -framework CoreGraphics -framework AppKit \
+//!      <SDL-distribution>/src/main/macosx/SDLMain.m "
+//!
+//! where <SDL-lib> is the directory holding the libSDL.a,
+//!       <SDL-include>/SDL is a directory holding headers like SDL.h
+//!   and <SDL-distribution> is, as above, the root of SDL 1.2 source download.
+
+#[no_main];
+
+extern crate native;
+extern crate sdl;
+
+use std::rand::Rng;
+use std::rand;
+
+#[no_mangle]
+pub extern "C" fn SDL_main(argc: int, argv: **u8) {
+    native::start(argc, argv, real_main);
+}
+
+pub fn real_main() {
+    sdl::init([sdl::InitVideo]);
+    sdl::wm::set_caption("rust-sdl demo - video", "rust-sdl");
+
+    let mut rng = rand::rng();
+    let screen = match sdl::video::set_video_mode(800, 600, 32, [sdl::video::HWSurface],
+                                                                [sdl::video::DoubleBuf]) {
+        Ok(screen) => screen,
+        Err(err) => fail!("failed to set video mode: {}", err)
+    };
+
+    // Note: You'll want to put this and the flip call inside the main loop
+    // but we don't as to not startle epileptics
+    for i in range(0, 10) {
+        for j in range(0, 10) {
+            screen.fill_rect(Some(sdl::Rect {
+                x: (i as i16) * 800 / 10,
+                y: (j as i16) * 600 / 10,
+                w: 800 / 10,
+                h: 600 / 10
+            }), rng.gen::<sdl::video::Color>());
+        }
+    }
+
+    screen.flip();
+
+    'main : loop {
+        'event : loop {
+            match sdl::event::poll_event() {
+                sdl::event::QuitEvent => break 'main,
+                sdl::event::NoEvent => break 'event,
+                sdl::event::KeyEvent(k, _, _, _)
+                    if k == sdl::event::EscapeKey
+                        => break 'main,
+                _ => {}
+            }
+        }
+    }
+
+    sdl::quit();
+}

